Update definition of science configurations

SeanBryan51 opened the pull request:
Move science configurations from site_configs.yaml to config.yaml. This change is a part of
https://github.com/CABLE-LSM/benchcab/pull/44

Specify `cable` in the top level of each science config dictionary. This change is a part of
https://github.com/CABLE-LSM/benchcab/pull/48

SeanBryan51 committed:
Update definition of science configurations

Move science configurations from site_configs.yaml to
config.yaml. This change is a part of
https://github.com/CABLE-LSM/benchcab/pull/44

Specify `cable` in the top level of each science config
dictionary. This change is a part of
https://github.com/CABLE-LSM/benchcab/pull/48

ccarouge commented on config.yaml:
```diff
@@ -42,3 +42,25 @@ modules: [
+
+# The 'science_configurations' key is an optional key for users to provide their
+# own custom science configurations. To run benchcab with the default set of science
+# configurations, remove this key from the config file.
```
I don't like that the user has to do something to run the default. The `bench_example` repo should be set so a user can run the default with minimum changes. Ideally, no changes.

SeanBryan51 replied:
Now that we are writing documentation for the config file, I think we can remove the optional keys (`revision`, `patch`, and `science_configurations`) as this will produce the default configurations. However, we should include a comment somewhere in the config file that makes the user aware that optional keys exist by adding a link to the documentation. What do you think?

ccarouge replied:
Yes, and have code examples in the documentation so people can copy-paste directly which I believe you have done.

SeanBryan51 committed:
Remove optional keys by default

The `bench_example` repo should be set so that a user can
run the default configuration with minimal changes.

Provide a link to the documentation in the config file
so that the user may find all available keys to be used
in the config file.

Addresses comments made by @ccarouge
https://github.com/CABLE-LSM/bench_example/pull/4#discussion_r1150065283
